src/dotnet/Chat.Service/ChatsMigrationBackend.cs

```diff
@@ -0,0 +1,449 @@
+            .ExecuteUpdateAsync(setters => setters
+                    .SetProperty(c => c.ChatId,
+                        c => newChatSid)
+                    .SetProperty(c => c.Id,
+                        c => c.Id.Replace(chatSid, newChatSid)),
+                cancellationToken)
+            .ConfigureAwait(false);
+#pragma warning restore CA1307
+
+        Log.LogInformation("Updated ChatId and Id for {Count} mention records", updateCount);
+    }
+
+    private async Task UpdateReactions(
+        ChatDbContext dbContext,
+        string chatSid,
+        string newChatSid,
+        IReadOnlyCollection<MigratedAuthor> migratedAuthors,
+        CancellationToken cancellationToken)
+    {
+        int updateCount;
+
+        var authorSids = await dbContext.Reactions
+            .Where(c => c.Id.StartsWith(chatSid))
+            .Select(c => c.AuthorId)
+            .Distinct()
+            .ToListAsync(cancellationToken)
+            .ConfigureAwait(false);
+
+        foreach (var authorSid in authorSids) {
+            var newAuthorId = migratedAuthors.First(c => OrdinalEquals(c.OriginalAuthor.Id.Value, authorSid)).NewId;
+            var newAuthorSid = newAuthorId.Value;
+#pragma warning disable CA1307
+            updateCount = await dbContext.Reactions
+                .Where(c => c.Id.StartsWith(chatSid))
+                .Where(c => c.AuthorId == authorSid)
+                .ExecuteUpdateAsync(setters => setters
+                        .SetProperty(c => c.Id,
+                            c => c.Id.Replace(authorSid, newAuthorSid))
+                        .SetProperty(c => c.AuthorId,
+                            newAuthorSid),
+                    cancellationToken)
+                .ConfigureAwait(false);
+#pragma warning restore CA1307
+            Log.LogInformation("Updated AuthorId for {Count} reaction records with old author id '{AuthorId}'."+
+                " New author id is '{NewAuthorId}'", updateCount, authorSid, newAuthorSid);
+        }
+
+#pragma warning disable CA1307
+        updateCount = await dbContext.Reactions
+            .Where(c => c.Id.StartsWith(chatSid))
+            .ExecuteUpdateAsync(setters => setters
+                    .SetProperty(c => c.EntryId,
+                        c => c.EntryId.Replace(chatSid, newChatSid))
+                    .SetProperty(c => c.Id,
+                        c => c.Id.Replace(chatSid, newChatSid)),
+                cancellationToken)
+            .ConfigureAwait(false);
+#pragma warning restore CA1307
+
+        Log.LogInformation("Updated EntryId and Id for {Count} reaction records", updateCount);
+    }
+
+    private async Task UpdateReactionSummaries(
+        ChatDbContext dbContext,
+        string chatSid,
+        string newChatSid,
+        IReadOnlyCollection<MigratedAuthor> migratedAuthors,
+        CancellationToken cancellationToken)
+    {
+        #pragma warning disable CA1307
+        var updateCount = await dbContext.ReactionSummaries
```

Comment:
Maybe makes sense to recompute these - i.e. delete the old ones & run a command (or commands) which will recompute the summaries.

Reply:
Why do't you like direct updates like it's implemented now?